### DISCUSSION TITLE
Burn function 🔥

### DISCUSSION THREAD

**Discussion by Cryptonian87** [hashlips-lab/nft-erc721-collection#264] [Ideas]
Hi,

Thank you for this amazing work that will turn up to be the ultimate Framework for deploying safely and surely an NFT collection.

My idea would be to create a burn function that can be executed as script i' order to burn the remaning supply.

👍

**Comment by ecerroni** [comment]
Not sure if this is this what you're actually looking for. However, on the smart contract to reduce the max supply to the current total supply adding the following function would be enough: 

```
  function reduceMaxSupplToCurrentSupply() public onlyOwner {
    uint256 supply = totalSupply();
    require((maxSupply - supply) > 0, "No action needed");
    maxSupply = supply;
  }
```

So let's say you open the public sale hoping it solds out, but at a certain point the number of people minting slows down and finally stops:
- 10.000 NFT offered
- 2.500 actually minted

Using the previous function in the smart contract would set the following:
- 2.500 NFT offered
- 2.500 actually minted
- minting is over

That way you can safe guar the buyers reducing the supply to what was minted until that point.

**Reply by ClubPEC** [reply]
But how to reduce supply to a new quantity? e.g 10K initial supply but due to lower demand want to reduce total supply to 6k.
